src/rsvc/srv.c

```diff
@@ -868,7 +852,7 @@ ds_rsvc_dist_start(enum ds_rsvc_class_id class, const uuid_t pool_uuid,
 		goto out;
 	in = crt_req_get(rpc);
 	in->sai_class = class;
-	uuid_copy(in->sai_svc_uuid, pool_uuid);
+	daos_iov_copy(&in->sai_svc_id, id);
```

Comment:
What about the daos_iov_free(&in->sai_svc_id)? :)

Reply:
Good catch. That was a pretty stupid error. Fixed!